### PR TITLE
feat!: remove support for `node10` resolution + cjs

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

closes https://github.com/nuxt/module-builder/pull/421

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This, at long last, removes support for legacy `node10` module resolution + CJS output

### CODE REVIEW

File: src/commands/build.ts [hunk 1]
```diff
@@ -1,11 +1,12 @@
 import { existsSync, promises as fsp } from 'node:fs'
 import { pathToFileURL } from 'node:url'
-import { basename, dirname, join, normalize, resolve } from 'pathe'
+import { basename, dirname, join, normalize, relative, resolve } from 'pathe'
```
Comment (danielroe) [inline src/commands/build.ts]:
```suggestion
import { basename, dirname, join, normalize, resolve } from 'pathe'
```

File: src/commands/build.ts [hunk 2]
```diff
@@ -71,7 +73,7 @@ export default defineCommand({
           target: 'esnext',
         },
         emitCJS: false,
-        cjsBridge: true,
+        cjsBridge: false,
```
Comment (pi0) [inline src/commands/build.ts:76]: This also disables ability to have __filename / __dirname from ESM files litle bit dangerous we should clearly document migration if we disable

**Review by pi0 — APPROVED** [review]

File: src/commands/build.ts (hunk 2 above)
Comment (pi0) [reply src/commands/build.ts:76]: Jiti in dev also allows this so modules only fail in production published

Comment (danielroe) [reply src/commands/build.ts:76]: will make sure to document ❤️

Comment (danielroe) [reply src/commands/build.ts:76]: didn't see the last comment - do you think it is better to keep?

Comment (pi0) [reply src/commands/build.ts:76]: i think made it few mins after when remembered. more streict eslint rules might help but that's just a fact that module authors might miss until too late. disabling is fine too considering we eventually want to drop it.

**Pull request merged**